ui/src/dataExplorer/components/SaveAsVariable.tsx

```diff
@@ -23,28 +25,38 @@ interface DispatchProps {
 
 interface StateProps {
   initialScript?: string
+  variables: Variable[]
 }
 
 type Props = StateProps & DispatchProps & OwnProps
 
 class SaveAsVariable extends PureComponent<Props & WithRouterProps> {
   render() {
-    const {onHideOverlay, onCreateVariable, initialScript} = this.props
+    const {
+      onHideOverlay,
+      onCreateVariable,
+      initialScript,
+      variables,
+    } = this.props
 
     return (
       <VariableForm
         onHideOverlay={onHideOverlay}
         onCreateVariable={onCreateVariable}
         initialScript={initialScript}
+        variables={variables}
       />
     )
   }
 }
 
 const mstp = (state: AppState): StateProps => {
   const activeQuery = getActiveQuery(state)
+  const variables = extractVariablesList(state)
```

Comment:
Note: variables state is loaded by the `DataExplorer`, which is a sibling. If we want to be more explicit we could lift that `GetResources` to the `DataExplorerPage`